plugins/woocommerce/src/Admin/API/Reports/Categories/DataStore.php

```diff
@@ -65,6 +65,13 @@ class DataStore extends ReportsDataStore implements DataStoreInterface {
 	 */
 	protected $context = 'categories';
 
+	/**
+	 * SQL columns to select in the db query.
+	 *
+	 * @var array
+	 */
+	protected $report_columns;
```

Comment:
There is l[ogic in the base data store](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Reports/DataStore.php#L132:L139) which depends on whether this property exists or not. If it's okay for that block of code to run against all the analytics data stores then `$report_columns` could be added to the base class instead of the individual classes.

Reply:
cc @chihsuan

Reply:
> If it's okay for that block of code to run against all the analytics data stores then $report_columns could be added to the base class instead of the individual classes.

Yep, I agree. I didn't inspect all datastores, though, but at least all the ones I changed did have the property.

Reply:
> If it's okay for that block of code to run against all the analytics data stores then $report_columns could be added to the base class instead of the individual classes.

Good point! I think it's okay to add it to the base class as an empty array.

Reply:
> I think it's okay to add it to the base class as an empty array.

@jorgeatorres That works for me.